bundles/org.openhab.binding.herzborg/src/main/java/org/openhab/binding/herzborg/internal/CurtainHandler.java

```diff
@@ -0,0 +1,224 @@
+                }
+                break;
+        }
+
+        if (pkt != null) {
+            final Packet p = pkt;
+            scheduler.schedule(() -> {
+                Packet reply = doPacket(p);
+
+                if (reply != null) {
+                    logger.trace("Function {} addr {} reply {}", p.getFunction(), p.getDataAddress(),
+                            DatatypeConverter.printHexBinary(reply.getBuffer()));
+                }
+            }, 0, TimeUnit.MILLISECONDS);
+        }
+    }
+
+    private Packet buildPacket(byte function, byte data_addr) {
+        return new Packet((short) config.address, function, data_addr);
+    }
+
+    private Packet buildPacket(byte function, byte data_addr, byte value) {
+        return new Packet((short) config.address, function, data_addr, value);
+    }
+
+    private Packet buildPacket(byte function, byte data_addr, int value) {
+        return buildPacket(function, data_addr, (byte) value);
+    }
+
+    @Override
+    public void initialize() {
+        Bridge bridge = getBridge();
+
+        if (bridge == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_INITIALIZING_ERROR, "Bridge not present");
+            return;
+        }
+
+        BridgeHandler handler = bridge.getHandler();
+
+        if (handler == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_INITIALIZING_ERROR, "Bridge has no handler");
+            return;
+        }
+
+        bus = ((BusHandler) handler).getBus();
+        config = getConfigAs(CurtainConfiguration.class);
+
+        updateStatus(ThingStatus.UNKNOWN);
+        logger.trace("Successfully initialized, starting poll");
+        pollFuture = scheduler.scheduleWithFixedDelay(this::poll, 1, config.poll_interval, TimeUnit.SECONDS);
+    }
+
+    @Override
+    public void dispose() {
+        stopPoll();
+    }
+
+    private void stopPoll() {
+        if (pollFuture != null) {
+            pollFuture.cancel(true);
+            pollFuture = null;
+        }
+    }
+
+    private @Nullable synchronized Packet doPacket(Packet pkt) {
+        Bus bus = this.bus;
+
+        if (bus == null) {
+            // This is an impossible situation but Eclipse forces us to handle it
+            logger.warn("No Bridge sending commands");
+            return null;
+        }
+
+        try {
+            Packet reply = bus.doPacket(pkt);
+
+            if (reply == null) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+                return null;
+            }
+
+            if (reply.isValid()) {
+                updateStatus(ThingStatus.ONLINE);
+                return reply;
+            } else {
+                logger.warn("Invalid reply received: {}", DatatypeConverter.printHexBinary(reply.getBuffer()));
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Invalid response received");
```

Comment:
```suggestion
                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Invalid response received:" + DatatypeConverter.printHexBinary(reply.getBuffer()));
```

When the thing status is changed, it already gets logged to the event.log and it is more visible in the main UI. To prevent a double up of logging, this way is preferred.